spec/controllers/transcriptions_controller_spec.rb

```diff
@@ -167,45 +167,66 @@
     end
 
     describe 'roles' do
-      before do
-        get :show, params: { id: transcription.id }
+      shared_examples 'lockable transcription' do
+        let(:locked_transcription) { create(:transcription, locked_by: 'kar-aniyuki', lock_timeout: (DateTime.now + 1.hours)) }
+
+        it 'locks the transcription if not already locked' do
+          get :show, params: { id: transcription.id }
+          expect(json_data['attributes']['locked_by']).to eq(user.login)
+        end
+
+        it 'does not lock the transcription if it is already locked' do
+          expect do
+            get :show, params: { id: transcription.id }
+          end.not_to change { locked_transcription.locked_by }
+        end
       end
 
       context 'without any roles' do
         let(:user) { create(:user, roles: {}) }
+
         it 'returns a 403' do
+          get :show, params: { id: transcription.id }
           expect(response).to have_http_status(:forbidden)
         end
       end
 
       context 'as an admin' do
         let(:user) { create(:user, :admin) }
+
         it 'returns the full scope' do
+          get :show, params: { id: transcription.id }
           expect(response).to have_http_status(:ok)
           expect(json_data).to have_id(transcription.id.to_s)
         end
 
-        it 'locks the transcription' do
-          expect(json_data['attributes']['locked_by']).to eq(user.login)
-        end
+        it_behaves_like 'lockable transcription'
       end
 
       context 'as an editor' do
-        let(:user) { create(:user, roles: { transcription.workflow.project.id => ['expert'] }) }
-
-        it 'locks the transcription' do
-          expect(json_data['attributes']['locked_by']).to eq(user.login)
+        let(:user) do
+          editor_roles =
```

Comment:
Why put this in the `let` block?

Reply:
ah, there wasn't a good reason to do so, I was just extracting this part from `create(:user, roles: editor_roles)`. Let me put that in it's own `let`.